airflow/utils/log/logging_mixin.py

```diff
@@ -116,6 +116,13 @@ def isatty(self):
         """
         return False
 
+    def fileno(self):
+        """
+        Returns the stdout file descriptor 1.
+        For compatibility reasons e.g python subprocess module stdout redirection.
+        """
+        return 1
```

Comment:
This is wrong in many cases:

- When the stream is stderr
- When stdout is redirected, if something wrote to fd 1 directly (which is the purpose of this method) then it would by-pass the redirection.

Reply:
Is there a way to identify if the stream is stderr?

That's true, the only alternative I can think of would be creating a pipe, returning its fd and writing everything written to that pipe.

I'm not sure if this would be an acceptable solution / workaround.